portfolio/src/main/java/com/google/sps/commentart/Painter.java

```diff
@@ -0,0 +1,35 @@
+package com.google.sps.commentart;
+
+import java.awt.Color;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** Painter determines the paints that should be used for painting. */
+public class Painter {
+  private final Map<String, Palette> paints = new HashMap<>(); 
+
+  /** Adds a {@code Palette} to {@code this). */
+  public void addPalette(Palette palette) {
+    paints.put(palette.getId(), palette);
+  }
+
+  /** 
+   * Updates the paint used by {@code Painter}.
+   * @param keyString Key that is associated with the {@code Palette} which will be affected.
+   * @param colour The colour used to update {@code Painter}.
+   */
+  public void update(String keyString, Color colour) {
+    Palette palette = paints.get(keyString);
+    palette.addColour(colour);
+  }
+
+  /** Gets all paints that are available. */
+  public Collection<Color> getPaints() {
```

Comment:
Generally Collection might be too general for this, you can find out the implementations of Collection at https://docs.oracle.com/javase/9/docs/api/java/util/Collection.html. I'm not sure if any of it would serve your purpose. For example, TreeSet, which may change your order within the collection when you convert it to list later.

Reply:
Hmm... True. This didn't occur to me. Thanks (: